src/main/java/org/springframework/data/ldap/repository/PagingLdapRepository.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2016-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.ldap.repository;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.ldap.query.LdapQuery;
+
+/**
+ * Paging repository for LDAP.
+ *
+ * @author Houcheng Lin
+ */
+public interface PagingLdapRepository<T> extends LdapRepository<T> {
```

Comment:
Any specific reason why you created a new interface instead of adding these methods to `LdapRepository`?